### PR TITLE
Adding MT Sample clinical note dataset

### PR DESCRIPTION
**Pull request by pruksmhc**
## Description
This creates the a dataset for clinical note generation from MT Samples, such as the below. 

```
User: Write a clinical note about a patient with the following chief complaint:  Stomach pain for 2 weeks.
ASSISTANT: 
CHIEF COMPLAINT: Stomach pain for 2 weeks. HISTORY OF PRESENT ILLNESS: The patient is a 45yo Mexican man without significant past medical history who presents to the emergency room with complaints of mid-epigastric and right upper quadrant abdominal pain for the last 14 days. The pain was initially crampy and burning in character and was relieved with food intake. He also reports that it initially was associated with a sour taste in his mouth. He went to his primary care physician who prescribed cimetidine 400mg qhs x 5 days; however, this did not relieve his symptoms. In fact, the pain has worsened such that the pain now radiates to the back but is waxing and waning in duration. It is relieved with standing and ambulation and exacerbated when lying in a supine position. He reports a decrease in appetite associated with a 4 lb. wt loss over the last 2 wks. He does have nausea with only one episode of non-bilious, non-bloody emesis on day of admission. He reports a 2 wk history of subjective fever and diaphoresis. He denies any diarrhea, constipation, dysuria, melena, or hematochezia. His last bowel movement was during the morning of admission and was normal. He denies any travel in the last 9 years and sick contacts.PAST MEDICAL HISTORY: Right inguinal groin cyst removal 15 years ago. Unknown etiology. No recurrence.
PAST SURGICAL HISTORY: Left femoral neck fracture with prosthesis secondary to a fall 4 years ago.
FAMILY HISTORY: Mother with diabetes. No history of liver disease. No malignancies.
SOCIAL HISTORY: The patient was born in central Mexico but moved to the United States 9 years ago. He is on disability due to his prior femoral fracture. He denies any tobacco or illicit drug use. He only drinks alcohol socially, no more than 1 drink every few weeks. He is married and has 3 healthy children. He denies any tattoos or risky sexual behavior.
ALLERGIES: NKDA.
MEDICATIONS: Tylenol prn (1-2 tabs every other day for the last 2 wks), Cimetidine 400mg po qhs x 5 days.
REVIEW OF SYSTEMS:  No headache, vision changes. No shortness of breath. No chest pain or palpitations.
PHYSICAL EXAMINATION: Vitals: T 100.9-102.7 BP 136/86 Pulse 117 RR 12 98% sat on room air,Gen: Well-developed, well-nourished, no apparent distress.HEENT: Pupils equal, round and reactive to light. Anicteric. Oropharynx clear and moist.Neck: Supple. No lymphadenopathy or carotid bruits. No thyromegaly or masses.CHEST: Clear to auscultation bilaterally.CV: Tachycardic but regular rhythm, normal S1/S2, no murmurs/rubs/gallops.Abd: Soft, active bowel sounds. Tender in the epigastrium and right upper quadrant with palpation associated with slight guarding. No rebound tenderness. No hepatomegaly. No splenomegaly.Rectal: Stool was brown and guaiac negative.Ext: No cyanosis/clubbing/edema.Neurological: He was alert and oriented x3. CN II-XII intact. Normal 2+ DTRs. No focal neurological deficit.Skin: No jaundice. No skin rashes or lesions.
IMAGING DATA:CT Abdomen with contrast ( 11/29/03 ):  There is a 6x6 cm multilobular hypodense mass seen at the level of the hepatic hilum and caudate lobe which is resulting in mass effect with dilatation of the intrahepatic radicals of the left lobe of the liver. The rest of the liver parenchyma is homogeneous. The gallbladder, pancreas, spleen, adrenal glands and kidneys are within normal limits. The retroperitoneal vascular structures are within normal limits. There is no evidence of lymphadenopathy, free fluid or fluid collections.HOSPITAL COURSE:  The patient was admitted to the hospital for further evaluation. A diagnostic procedure was performed.
```

## Issue
https://github.com/LAION-AI/Open-Assistant/issues/721

## Checks
[X] This dataset doesn't contain PHI

### CODE REVIEW

**Comment by pruksmhc** [comment]
I may work on adding Synthea as a source in addition to MT Samples since MT samples is very small:
https://github.com/synthetichealth/synthea 
It seems like this source may also have notes:
https://www.eurorad.org/teaching-cases

**Comment by ontocord** [comment]
This looks really good! We use the separator \n\nRosey: for our assistant instead of "ASSISTANT:" but this can be easily fixed.
Did you run a PII checker on this so no names are put in there?

**Comment by pruksmhc** [comment]
Ah got it. @ontocord just for curiosity, is Rosey the name of the assistant?

MT Samples is made up - on their website, they state
"NOTE: These transcribed medical transcription sample reports and examples are provided by various users and are for reference purpose only. MTHelpLine does not certify accuracy and quality of sample reports. These transcribed medical transcription sample reports may include some uncommon or unusual formats; this would be due to the preference of the dictating physician. All names and dates have been changed (or removed) to keep confidentiality. Any resemblance of any type of name or date or place or anything else to real world is purely incidental."

**Comment by ontocord** [comment]
Ok re PII then. add the change to put in "\n\nRosey:" if you have time. Otherwise, we will do this in post processing. Thank you!

**Comment by ontocord** [comment]
Yes - Rosey is an internal name of the assitant. Not official.

File: openassistant/datasets/mt_note_generation/hub.py [hunk 1]
```diff
@@ -0,0 +1,21 @@
+from dataclasses import dataclass
```
Comment (ontocord) [inline openassistant/datasets/mt_note_generation/hub.py:1]: Really good idea to include dataclass

File: openassistant/datasets/mt_note_generation/prepare.py [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+    mt_note_subset = mt_dataset[mt_dataset.apply(filter_for_notes, axis=1)]
+    return mt_note_subset
+
+
+def is_chief_complaint(section):
+    return "chief complaint" in section.lower()
+
+
+def get_conversations(dataset):
+    def normalize_transcript(x):
```
Comment (ontocord) [inline openassistant/datasets/mt_note_generation/prepare.py:33]: I noticed that sometimes the output contains straggline commas or two periods. maybe we can clean that up. Should be simple. Just break up the data by "." and then remove single character sentences, etc and then join it all by "." again.

File: openassistant/datasets/mt_note_generation/prepare.py [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+import random
+import re
+import sys
+from string import punctuation
+
+import kaggle
+import pandas as pd
+
+CLINICAL_NOTE_GENERATION_TEMPLATE = """User: Write a clinical note about a patient with the following {section}: {section_information}.
+ASSISTANT: {note}"""
```
Comment (ontocord) [inline openassistant/datasets/mt_note_generation/prepare.py:13]: Change to "Assistant" or "Rosey" please.

**Review by ontocord — APPROVED** [review]
These are small nits. We can merge without them, but if yp has time, that would be awesome to have the changes.

**Review by ontocord — APPROVED** [review]
These are small nits. We can merge without them, but if @pruksmhc  has time, that would be awesome to have the changes.

**Review by andreaskoepf — APPROVED** [review]
OK, nice job! @pruksmhc  if you have time to address @ontocord 's comments that would be nice. Otherwise we will merge it as it is in ~24h.

File: openassistant/datasets/mt_note_generation/prepare.py (hunk 3 above)
Comment (pruksmhc) [reply openassistant/datasets/mt_note_generation/prepare.py]: done

File: openassistant/datasets/mt_note_generation/prepare.py (hunk 2 above)
Comment (pruksmhc) [reply openassistant/datasets/mt_note_generation/prepare.py:33]: done

**Comment by andreaskoepf** [comment]
@pruksmhc thanks a lot! :-)

**Pull request merged**